mpvScripts.{blacklistExtensions, seekTo}: Init at unstable-2022-10-02

nicoonoclaste opened the pull request:
###### Description of changes

- Add nicoo to `lib.maintainers`
- `mpvScripts.seekTo`: Init at 2022-10-02

`seek-to.lua` is a 3rd-party mpv script that enables users to seek to a
specific timestamp, rather than manually go to some position "close enough"
to where they meant to be.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

nicoonoclaste committed:
mpvScripts.seekTo: init at unstable-2022-10-02

kilianar commented on maintainers/maintainer-list.nix:
```diff
@@ -10692,6 +10692,15 @@
   };
+  nicoo = {
+    email = "nicoo@debian.org";
+    gihub = "nbraud";
```
```suggestion
    github = "nbraud";
```

nbraud replied:
Thanks for catching that, was still trying to interpret ofborg's output  :sweat_smile:

nbraud commented:
Trivial rebase to solve conflict in `all-packages.nix`

kilianar commented:
I have made some improvements to your code. Please review and make any further changes to your pull request as you see fit.

You do not need to use `let` to define `src` and `version`. Additionally, there were several unused arguments including `fetchpatch`, `python3`, and `nix-update-script`.

Since we only need one file from the repository, we could use a sparse checkout. However, the repository is quite small so I'm unsure if it's worth the effort.

If a package is not a release but a commit from a repository, then the version attribute must be the date of that (fetched) commit. The date must be in "unstable-YYYY-MM-DD" format.  See https://nixos.org/manual/nixpkgs/unstable/#sec-package-naming for reference.

```nix
{
  lib
, stdenvNoCC
, fetchFromGitHub
}:

# Usage: `pkgs.mpv.override { scripts = [ pkgs.mpvScripts.seekTo ]; }`
stdenvNoCC.mkDerivation {
  pname = "mpv_seek-to";
  version = "unstable-2022-10-02";
  
  src = fetchFromGitHub {
      owner = "occivink";
      repo = "mpv-scripts";
      rev = "af360f332897dda907644480f785336bc93facf1";
      hash = "sha256-Rp795c3c73sT4Ws8PhzeQlSC9U6+mWcWe/Wtm8XzPaM=";
      sparseCheckout = [
        "scripts/seek-to.lua"
      ];
  };

  dontBuild = true;

  installPhase = ''
    install -Dm644 scripts/seek-to.lua $out/share/mpv/scripts/seek-to.lua
  '';

  passthru.scriptName = "seek-to.lua";

  meta = with lib; {
    description = "Mpv script for seeking to a specific position";
    homepage = "https://github.com/occivink/mpv-scripts";
    license = licenses.unlicense;
    platforms = platforms.all;
    maintainers = with maintainers; [ nicoo ];
  };
}

```

nbraud commented:
> You do not need to use `let` to define `src` and `version`.

Yes, I did that because I was considering packaging more scripts from the same repo, in which case factoring out `src` and `version` is useful.

> Additionally, there were several unused arguments including `fetchpatch`, `python3`, and `nix-update-script`.

Thanks, I missed this while cleaning up.

> Since we only need one file from the repository, we could use a sparse checkout. However, the repository is quite small so I'm unsure if it's worth the effort.

Same (unsure it's worth the effort) ; I also considered fetching that one file directly from Github from a link like so:
```
https://github.com/occivink/mpv-scripts/raw/${commit}/scripts/seek-to.lua
```

All in all, it still seemed excessive given the smallness of the repo, and the maintainability aspects.

> If a package is not a release but a commit from a repository, then the version attribute must be the date of that (fetched) commit. The date must be in "unstable-YYYY-MM-DD" format. See https://nixos.org/manual/nixpkgs/unstable/#sec-package-naming for reference.

OK, I looked for something like that in the reference but didn't think to check under "package naming" since version naming conventions :sweat_smile:

nbraud commented:
@kilianar Updated the PR; as mentioned above, I kept `src` and `version` factored out, as it's likely I will package other scripts from that repo.

Sorry for letting it sit for a week, I expected to get it done ~immediately but the fallout from another PR took priority.

mkg20001 commented on pkgs/applications/video/mpv/scripts/seek-to.nix:
```diff
@@ -0,0 +1,30 @@
+{ lib, stdenvNoCC, fetchFromGitHub }:
+
+let src = fetchFromGitHub {
+      owner = "occivink"; repo = "mpv-scripts";
+      rev = "af360f332897dda907644480f785336bc93facf1";
+      hash = "sha256-KdCrUkJpbxxqmyUHksVVc8KdMn8ivJeUA2eerFZfEE8=";
+    };
+    version = "unstable-2022-10-02";
+in
```
```suggestion
let
  src = fetchFromGitHub {
    owner = "occivink";
    repo = "mpv-scripts";
    rev = "af360f332897dda907644480f785336bc93facf1";
    hash = "sha256-KdCrUkJpbxxqmyUHksVVc8KdMn8ivJeUA2eerFZfEE8=";
  };
  version = "unstable-2022-10-02";
in
```

mkg20001 replied:
> Yes, I did that because I was considering packaging more scripts from the same repo, in which case factoring out src and version is useful.

You could create a occivink-mpv-generic.nix with a function that takes pname, etc. in this pr

nbraud replied:
Done!  It seemed a bit silly to have it all in separate 2-line files, so I did something slightly different (and feedback is much appreciated ;3)

mkg20001 replied:
please move the generic function into it's own file

do something like (don't use name as the argument name since that might be set to something...)

```
{ scriptName
, stdenvNoCC
}:

...mkDerivation rec {
}
```

and then just callPackage ./occivik-generic.nix { scriptName = "bla"; }  it in toplevel if it's really just the scriptName (otherwise create a seperate file if it's more (like extra patches) "{ callPackage }: callPackage ./occivik-generic.nix { scriptName = "...."; other = ... }```)

the unionOfDisjoint in the other file: doing it like this is weird and I'm unsure if it's side-effect free (so do it in toplevel)

or otherwise creating a scope like cinnamon does it https://github.com/NixOS/nixpkgs/blob/master/pkgs/desktops/cinnamon/default.nix#L3 with all the packages is fine by me. (that's how that usually gets handled)

or feel free to look at existing code in nixpkgs and take inspiration

mkg20001 replied:
I don't know what the nix/nixpkgs behaviour will be but if you still want to pass name, etc you could perhaps `{ stdenv... } @ args:` and then merge args into that. but not sure if it's going to pull all of nixpkgs in then.

nbraud replied:
> and then just callPackage ./occivik-generic.nix { scriptName = "bla"; } it in toplevel if it's really just the scriptName 

Just the script name and description, yes.  The latter is what I didn't want to add to `all-packages.nix`, since almost nobody else did.

> the unionOfDisjoint in the other file: doing it like this is weird and I'm unsure if it's side-effect free (so do it in toplevel)

OK, replaced with `//`.

> or otherwise creating a scope like cinnamon does [...]with all the packages is fine by me. (that's how that usually gets handled)

I'm also fine making a scope if that's preferred.
Thanks a bunch for all the helpful feedback  <3

mkg20001 reviewed (changes requested):
>Add nicoo to lib.maintainers

should be `maintainers: add nicoo`

> mpvScripts.seekTo: Init at 2022-10-02
> mpvScripts.seekTo: Remove unnecessary inputs 
> mpvScripts.seekTo: 2022-10-02 → unstable-2022-10-02 

Maybe squash to "mpvScripts.seekTo: init at unstable-2022-10-02", so we don't have too many commits (if you add the generic, that could be a seperate commit though)

mkg20001 commented on pkgs/applications/video/mpv/scripts/seek-to.nix:
```diff
@@ -0,0 +1,30 @@
+{ lib, stdenvNoCC, fetchFromGitHub }:
```
```suggestion
{ lib
, stdenvNoCC
, fetchFromGitHub
}:
```

nicoonoclaste committed:
mpvScripts.occivink: Add generic helper for all scripts in repo

nicoonoclaste committed:
mpvScripts.blacklistExtensions: init at unstable-2022-10-02

nicoonoclaste committed:
mpvScripts.occivink: Minor fixes to the generic helper

- Renamed 'name' parameter to avoid collisions with mkDerivation's one.
- Ensured the passed parameters can always override generic defaults.

nicoonoclaste committed:
mpvScripts.occivink: make FDOs (Fixed-Output Derivations)
